heron/common/src/cpp/network/baseserver.cpp

```diff
@@ -49,8 +49,8 @@ sp_int32 BaseServer::Start_Base() {
   errno = 0;
   listen_fd_ = socket(options_.get_socket_family(), SOCK_STREAM, 0);
   if (listen_fd_ < 0) {
-    LOG(ERROR) << "Opening of a socket failed in server " << errno << "\n";
-    return -1;
+    PLOG(ERROR) << "Opening of a socket failed in server";
+    return errno;
```

Comment:
Are errno's negative? Otherwise it might not work. Instead I would suggest to return NOT_OK.